lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1298,45 +1314,40 @@ def _process_logging_relation_changed(self, relation: Relation):
         if self._charm.unit.is_leader():
             relation.data[self._charm.app].update(self._promtail_binary_url)
             logger.debug("Saved promtail binary url: %s", self._promtail_binary_url)
-            self.update_endpoint(relation)
+
+        self.update_endpoint(relation=relation)
 
         if relation.data.get(relation.app).get("alert_rules"):
             logger.debug("Saved alerts rules to disk")
             self._remove_alert_rules_files(self.container)
             self._generate_alert_rules_files(self.container)
             self._check_alert_rules()
 
-    def _endpoints(self) -> List[dict]:
-        """Return a list of Loki Push Api endpoints."""
-        endpoints = []
-        path = "/loki/api/v1/push"
-
-        if self._charm.ingress_per_unit.urls:
-            return [
-                {"url": url.rstrip("/") + path}
-                for url in self._charm.ingress_per_unit.urls.values()
-            ]
-
-        peers = [*self._charm.meta.peers.keys()][0]
-        units = self._charm.model.get_relation(peers).units
-        units.add(self._charm.unit)
-
-        for unit in units:
-            unit_number = unit.name.split("/")[-1]
-            endpoints.append({"url": self._url(unit_number)})
-
-        return endpoints
+    @property
+    def _url(self) -> str:
+        """Get local Loki Push API url.
 
-    def _url(self, unit_number) -> str:
-        """Get the url for a given unit."""
-        return "http://{}-{}.{}-endpoints.{}.svc.cluster.local:{}/loki/api/v1/push".format(
+        Returns: str
+        """
+        return "http://{}-{}.{}-endpoints.{}.svc.cluster.local:{}".format(
             self._charm.app.name,
-            unit_number,
+            self._charm.unit.name.split("/")[-1],
             self._charm.app.name,
             self._charm.model.name,
             self.port,
         )
 
+    def _endpoint(self, url) -> dict:
```

Comment:
I wish this would be a property rather than a method. Passing the `url` as argument feel unnecessary.